ts/javascript/vite.ts

```diff
@@ -0,0 +1,26 @@
+import { Executable } from "../executable.ts";
+import { Package } from "../package.ts";
+import { Plugin } from "../project.ts";
+
+export const plugin: Plugin<
+  { build: Package; dev: Executable; preview: Executable },
+  { node_modules: Package }
```

Comment:
It seems unfortunate that you have to have a `node_modules` package - that doesn't seem like something end users should have to care about, and it just pollutes the API. Maybe this isn't in scope for this PR, but it feels to me like, if all we're doing in mkNpmProject is creating the node_modules dir, we *shouldn't* have it create a `Package` at all (does anyone really want to build that?). In the Environment we *do* build it, and set a var like NODE_MODULES to the location, and then below you don't need to depend on `base.node_modules`, but can instead just `export PATH=$NODE_MODULES/bin:$PATH`

Reply:
I think that's a really good idea. It would mean that we have to change how `Environment`s work. Which is something we should do, but maybe this should be deferred for now?

The details are that currently we *don't* have a way of setting an env var for `Executable`s. Just for `Package`s and `Check`s. So the solution with an env var (say `NODE_MODULES`) would only work for the `build` field, not for `dev` and `preview`. We could of course just remove those two, since this ticket is mostly about bundling. I've done that in here: dc08eca0

What do you think? Also, @alexdavid, what do you think about this? Maybe we should not merge that commit and rather re-think `Environment`s separately?

Reply:
@alexdavid and I brainstormed a bit about `Environment`s and how to make them better. I think we should do this after this PR.

Reply:
Ok, I've removed that commit.